### ISSUE TITLE
Why metric inherits torch.nn.Module? 

### ISSUE THREAD

**Issue by Maybewuss** [PyTorchLightning/pytorch-lightning#4483]
If i define a complex metric and the computation may convert torch.tensor to numpy. Can this metric use in ddp？
If i want train model with train_dataloader in ddp and validate val_dataloader in single model with customized metric, how to do it?

**Comment by s-rog** [comment]
First question: yes, but you will have to convert the output within the metric back to tensor and cast it to the correct device.

Second question: check out our examples and documentation!

For future questions please consider using our forums! https://forums.pytorchlightning.ai/

**Comment by Maybewuss** [comment]
> First question: yes, but you will have to convert the output within the metric back to tensor and cast it to the correct device.
> 
> Second question: check out our examples and documentation!
> 
> For future questions please consider using our forums! https://forums.pytorchlightning.ai/

Second question: I have read the documents, the example of multi-gpu training code is `trainer = Trainer(gpus=8, accelerator='ddp')`, and i don't know what abstraction is used in validation step. In other words, whether validation will be called in ddp when local_rank equals zero and i do not need to think how to reduce metric from different opus?

**Comment by s-rog** [comment]
Validation step is pretty much exactly the same as training step besides using a different dataloader, `.eval()` and `no_grad()`

As such, it is called on all ranks in ddp. I would suggest trying out an existing metric so you get a feel for how they work before implementing your own.

**Comment by Maybewuss** [comment]
> Validation step is pretty much exactly the same as training step besides using a different dataloader, `.eval()` and `no_grad()`
> 
> As such, it is called on all ranks in ddp. I would suggest trying out an existing metric so you get a feel for how they work before implementing your own.

If i use self.log() in validation_epoch_end in ddp, need i to judge the local_rank to prevent log in different devices?
